Tests/Svg.UnitTests/SvgElementDeepCopyTest.cs

```diff
@@ -0,0 +1,126 @@
+            CheckDeepCopyInstance(new SvgSwitch());
+            CheckDeepCopyInstance(new SvgSymbol());
+            CheckDeepCopyInstance(new SvgTitle());
+            CheckDeepCopyInstance(new SvgUse());
+            CheckDeepCopyInstance(new SvgForeignObject());
+            CheckDeepCopyInstance(new SvgFilter());
+            CheckDeepCopyInstance(new SvgColourMatrix());
+            CheckDeepCopyInstance(new SvgGaussianBlur());
+            CheckDeepCopyInstance(new SvgMerge());
+            CheckDeepCopyInstance(new SvgMergeNode());
+            CheckDeepCopyInstance(new SvgOffset());
+            CheckDeepCopyInstance(new SvgGradientStop());
+            CheckDeepCopyInstance(new SvgLinearGradientServer());
+            CheckDeepCopyInstance(new SvgMarker());
+            CheckDeepCopyInstance(new SvgPatternServer());
+            CheckDeepCopyInstance(new SvgRadialGradientServer());
+            CheckDeepCopyInstance(new SvgPath());
+            CheckDeepCopyInstance(new SvgFont());
+            CheckDeepCopyInstance(new SvgFontFace());
+            CheckDeepCopyInstance(new SvgFontFaceSrc());
+            CheckDeepCopyInstance(new SvgFontFaceUri());
+            CheckDeepCopyInstance(new SvgGlyph());
+            CheckDeepCopyInstance(new SvgVerticalKern());
+            CheckDeepCopyInstance(new SvgHorizontalKern());
+            CheckDeepCopyInstance(new SvgMissingGlyph());
+            CheckDeepCopyInstance(new SvgText());
+            CheckDeepCopyInstance(new SvgTextPath());
+            CheckDeepCopyInstance(new SvgTextRef());
+            CheckDeepCopyInstance(new SvgTextSpan());
+        }
+
+        private void CheckDeepCopyInstance<T>(T src)
+            where T : SvgElement
+        {
+            var dest = src.DeepCopy();
+            Assert.IsInstanceOf<T>(dest);
+        }
+
+        /// <summary>
+        /// Tests that the deep copy of a <see cref="SvgText"/> is done correctly where the
+        /// text element has contains only text and now other elements like <see cref="SvgTextSpan"/>.
+        /// </summary>
+        [Test]
+        public void TestSvgTextElementDeepCopy()
+        {
+            var svgDocument = OpenSvg(GetResourceXmlDoc(GetFullResourceString(PureTextElementSvg)));
+            CheckDocument(svgDocument);
+
+            var deepCopy = (SvgDocument)svgDocument.DeepCopy<SvgDocument>();
+            CheckDocument(deepCopy);
+        }
+
+        /// <summary>
+        /// Checks the document if it contains the correct information when exported to XML.
+        /// </summary>
+        /// <param name="svgDocument">The SVG document to check.</param>
+        private static void CheckDocument(SvgDocument svgDocument)
+        {
+            //TODO: check if the assignablefrom indeed is the correct replacement
+            Assert.AreEqual(2, svgDocument.Children.Count);
+            Assert.IsAssignableFrom(typeof(SvgDefinitionList), svgDocument.Children[0]);
```

Comment:
Shouldn't that be rather a type test? e.g. something like:
```C#
Assert.That(svgDocument.Children[0], Is.InstanceOf(typeof(SvgDefinitionList)));
```

Reply:
@mrbean-bremen `IsAssignableFrom` is a type check. I believe the main difference between the two options is whether derived types also classify for the check.

Also, I would try to keep away from the `Assert.That` for readability. Most of the assertions are available directly from the Assert class, using an Assert.That (in my opinion) might impact readability for some developers.

Reply:
Ok, thanks! I think the readability of the `Assert.That`is a matter of taste, but I agree that we should stuck to one style.
I'll merge then, thanks @H1Gdev!